crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1674,6 +1674,16 @@ impl<'db> TypeInferenceBuilder<'db> {
             TargetKind::Name => value_ty,
         };
 
+        // NOTE: typing.Any is _not_ currently defined as a _SpecialForm in the typeshed, so we
+        // can't detect it in the same was a typing.NoReturn and friends.  If the definition in the
+        // typeshed changes, this heuristic will need to change as well.
```

Comment:
I'd also be inclined to leave out this comment, since I think typeshed is unlikely to change in this area (there's no strong motivation for the change, and it could cause issues for other type checkers). If typeshed were to change, our tests would alert us to the problem.
```suggestion
```